README.md

```diff
@@ -5,3 +5,52 @@
 Maybe it'll help if you think of them instead of redirect`atob`s. Basically redirectatob is a webpage that you pass a URL to in the location hash (i.e. domain/page.html#hash, but encoded with btoa. The webpage will convert the encoded hash back into a URL using atob through the client's JavaScript. I made it so that I could give this URL - http://www.woodus.com/den/games/dq4ds/mugshots.php - to a friend of mine. The site's blacklist was preventing me from posting the link, and, as it had happened to me multiple times before, I decided I needed to make a redirect service.
 
 I don't own a server, though, so just using the client's `btoa`/`atob` worked.
+
+Well, that's mostly the story... but for any developers who wish to generate these redirectatobs automatically, I encourage you to read on.
+
+# Specification
+
+A redirectabob is a URL which is under the `redirectatob.github.io` host, referencing the `index.html` file in the root directory; and, including a valid hash string which encodes the data for where the redirectatob should send the user when they load the page. It is highly recommended that the redirectatob will contain a forward slash after the host part, as ommiting it may cause an additional redirect to occur before the user is sent to their final destination.
+
+## Encoding destination URLs into hashes
+
+Usually, the hash is simply the `btoa()` JavaScript function run on the destination URL. Sometimes, however, this is not possible.
+
+### Valid schemas
+
+Any destination URL must specify a schema, and that schema must be one of
+
+- http
+- https
+- data
+
+You should not attempt to encode any other schema, as that will make the redirectatob invalid.
+
+### Encoding a URL
+
+#### In the Latin1 range
+
+If a destination URL contains NO characters OUTSIDE the Latin1 character range, then it should be encoded through the following steps
+
+1) Base64 encode the URL, in a way that replicates JavaScript's `btoa()` function.
```

Comment:
This is just one step, so you may as well get rid of "through the following steps"; replace these two lines with:

>If a destination URL contains NO characters OUTSIDE the Latin1 character range, then it should be Base64-encoded, in a way that replicates JavaScript's `btoa()` function.

Reply:
I think protocol refers to like "Now send this packet and do that"; whereas HTTP is the set of characters at the beginning of the URL which tell the browser which protocol to use. Okay, will fix -- I just like the two being symmetric